src/erasure-code/isa/ErasureCodeIsa.cc

```diff
@@ -117,33 +117,39 @@ int ErasureCodeIsa::decode_chunks(const set<int> &want_to_read,
 // -----------------------------------------------------------------------------
 
 void
-ErasureCodeIsa::isa_xor(char **data, char **coding, int blocksize)
+ErasureCodeIsa::isa_xor(char **data, char *coding, int blocksize, int data_vectors)
 {
-    // If addresses are aligned to 32 bytes, then we can use xor_gen()
-    // Otherwise, use byte_xor()
-    int i;
-    bool src_aligned = true;
+  char * xor_bufs[data_vectors+1];
```

Comment:
Variable-length arrays are a non-standard GNU extension; they exist only in C, but turned out to be a bad idea, and the C people have deprecated them. I'd rather not add new code to Ceph which uses VLA. (And even less so with a signed integer.)

Reply:
Thanks for taking a look Max. There are quite a few examples of VLAs in the Ceph plugin interface that have been there for a long time e.g.

```
int ErasureCodeIsa::encode_chunks(const set<int> &want_to_encode,
                                  map<int, bufferlist> *encoded)
{
  char *chunks[k + m];
```

```
int ErasureCodeIsa::decode_chunks(const set<int> &want_to_read,
                                  const map<int, bufferlist> &chunks,
                                  map<int, bufferlist> *decoded)
{
  unsigned blocksize = (*chunks.begin()).second.length();
  int erasures[k + m + 1];
  int erasures_count = 0;
  char *data[k];
  char *coding[m];
```

So I don't think using a VLA here would be any worse than what we already have. data_vectors is never going to be greater than k, so I don't think there is a new overflow risk here either. 

Do you have any suggestions for a nicer method given we need to pass a void ** to xor_gen() ?